File: angular/src/app/shared/dtos/signin.dto.ts
export interface SignInReqDTO {
  emailOrUserName: string;
  password: string;
}

export interface SignInResDTO {
  token?: string;
  requiresTwoFactor?: boolean;
  twoFactorToken?: string;
}

export interface TwoFactorSignInReqDTO {
  twoFactorToken: string;
  twoFactorCode: string;
}

export interface TwoFactorSignInResDTO {
  token: string;
}

File: backend/GofishApi/Dtos/SignInDTO.cs
using System.ComponentModel.DataAnnotations;

namespace GofishApi.Dtos;

public record SignInReqDTO(
    string EmailOrUserName,
    string Password
);

public record SignInResDTO(
    string? Token           = null,
    bool? RequiresTwoFactor = null,
    string? TwoFactorToken  = null
);

public record TwoFactorSignInReqDTO(
    [Required] string TwoFactorToken,
    [Required] string TwoFactorCode
);

public record TwoFactorSignInResDTO(
    string Token
);

public record ExternalSignInResDTO(
    string Token
)
{ }
